leetcode/54_spiral_matrix.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+class Solution:
+    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
+        """Returns the elements of the given 2D matrix in spiral order.
+
+        Args:
+            matrix (List[List[int]]): A 2D list representing the input matrix.
+
+        Returns:
+            List[int]: A list of integers representing the matrix elements in spiral order.
+        """
+
+        m, n = len(matrix), len(matrix[0])
+        r, c = 0, 0
+        UP, DOWN, RIGHT, LEFT = 0, 1, 2, 3
```

Comment:
```
UP = (-1,0)
DOWN = (1,0)
LEFT = (0,-1)
RIGHT = (0,1)
DIRECTIONS = [RIGHT, DOWN, LEFT, UP]

cur_dir = 0
if change_direction:
    cur_dir = (1 + cur_dir) % 4
```

Reply:
Increment by 1 step every iteration of the outer while loop.

Reply:
while I haven't processed all the elements

1. Move in the current direction. 
2. If I can't, change direction.

You can use a helper function to check if you're in bounds. Is my next position in the grid?